lib/cf_deployer/driver/auto_scaling_group.rb

```diff
@@ -50,6 +50,18 @@ def instance_statuses
         instance_info
       end
 
+      def wait_for_desired_capacity
+        Timeout::timeout(@timeout){
+          until desired_capacity_reached?
+            sleep 15
+          end
+        }
+      end
+
+      def desired_capacity_reached?
+        healthy_instance_count >= desired_capacity
```

Comment:
2) yes, we account for changes to the desired capacity by querying the ASG directly with each round of 'wait'.  Yes, the sdk client will query the ASG directly (this is default behavior, and there's actually a separate wrapper call to force the client to return cached responses instead of actually querying), and I confirmed this by running the client in irb, and confirmed by tweaking the desired value during a live deployment with a new version of this gem.